api/decode.py

```diff
@@ -0,0 +1,196 @@
+from flask import Flask, abort, json, jsonify
+import hashlib
+import pybitcointools
+from decimal import *
+from rpcclient import *
+
+app = Flask(__name__)
+app.debug = True
+
+@app.route('/<rawhex>')
+def decode_handler(rawhex):
+  return jsonify(decode(rawhex))
+
+
+def decode(rawhex):
+
+  transaction = decoderawtransaction(rawhex)['result']
+  rawtx = getrawtransaction(transaction['vin'][0]['txid'])
+  senders  = rawtx['result']['vout'][transaction['vin'][0]['vout']]['scriptPubKey']['addresses']
+  inputBTC = rawtx['result']['vout'][transaction['vin'][0]['vout']]['value']
+  reference = senders[0]
+
+  #get all multisigs
+  multisig_output = []
+  dest=""
+  for output in transaction['vout']:
+    if output['scriptPubKey']['type'] == 'multisig':
+      multisig_output.append(output) #grab msigs
+    else:
+      try:
+        for address in output['scriptPubKey']['addresses']:
+          if address not in (['1EXoDusjGwvnjZUyKkxZ4UHEf77z6A5S4P','mpexoDuSkGGqvqrkrjiFng38QPkJQVFyqv']+senders):
```

Comment:
One more note: only the first sender is ignored as potential receiver *and* only pay-to-pubkey-hash or pay-to-script-hash outputs qualify as receiver, whereby other outputs are permitted. This is different to the input rules where any other script type as input invalidates the transaction.

[The rules below the encoding image starting with "a single transaction must be constructed satisfying the following requirements"](https://github.com/mastercoin-MSC/spec#class-b-transactions-also-known-as-the-multisig-method) cover everything, as far as I can see.